BAM-22413 Update publisher reporting docs (#188)
hellonarrativ/api-docs: docs/pubreport.rst

Publisher Report
================

The Publisher Reports consist of the Clicks, Orders and Returns reports.
These reports contain detailed information about clicks, purchases, returns, attribution, and
earnings.

> The reports can be published to an FTP/SFTP location/S3/GCP bucket accessible by the partner.

Reviewer: ```suggestion
The reports can be published to an FTP/SFTP location or cloud storage bucket (S3/GCS) accessible by the partner.
```

Publisher Report
================

The Publisher Reports consist of the Clicks, Orders and Returns reports.

These reports contain detailed information about clicks, purchases, returns, attribution, and
earnings.

The reports can be published to an FTP/SFTP location or cloud storage bucket (S3/GCS) accessible by the partner.

> The reports is published on configurable interval which defaults to 24 hours.

Reviewer: ```suggestion
The reports are published on configurable interval which defaults to 24 hours.
```

Each interval publishes data for the last 7 days.

The naming convention for each of the reports is:

* Prefix: ``/<key>/YYYY/MM/`` (defaults to slug)

* Clicks: ``narrativ_log_files_clicks_YYYYMMDD.csv``. The date in the filename indicates the clicks in the report took place on that date

* Orders: ``narrativ_log_files_orders_YYYYMMDD.csv``. The date in the filename indicates the orders/checkouts in the report took place on that date

> * Returns: ``narrativ_log_files_orders_YYYYMMDD.csv``. The date in the filename indicates the returns in the report were processed on that date

Reviewer: ```suggestion
* Returns: ``narrativ_log_files_returns_YYYYMMDD.csv``. The date in the filename indicates the returns in the report were processed on that date
```

click_id                     Unique identifier for each click event attributed to this checkout
click_timestamp              Date time in GMT of the click attributed to this checkout
===========================  ===========

Returns
^^^^^^
The Returns File contains information about returns processed on that day from orders (checkouts) attributed
to the publisher at specific merchant partners in the past. This report operates at a day level and captures the
returns processed on a particular day, regardless of when the original checkout occurred. The returns can be partial

> or full, and can be broken over multiple days.

Reviewer: ```suggestion
or full, and can be broken over multiple days.

```

The report contains unique rows for each product returned in the order as they are processed,
even if multiple products from the same checkout are returned and processed separately. Each row will include
information about the number of items returned(quantity), the amount of earnings borne from the items returned,

> and the revenue associated with the returned items.

Reviewer: ```suggestion
and the revenue associated with the returned items.

```

Since the data in these reports pertains to returns, the earnings, revenue and quantity values are negative.

To gain a comprehensive view of returns, we recommend aggregating the values for each unique combination of
checkout_id, merchant and product_id (denoted by the order_product_id column) across all return files.

Since this report is designed to be used to complement the data in the orders reports, the column 'order_product_id'
can be used to join to the orders report.

This table can be used to identify -
* Quantity of products returned
* Earnings and revenue from the returned products

*What to expect in the Returns Report*

> Let's break it down with an example:

Reviewer: ```suggestion
Let's break it down with an example:

```

> Checkout 123 is attributed to a publication, and originally included :

Reviewer: ```suggestion
Checkout 123 is attributed to a publication, and originally included :

```

click_id                     Unique identifier for each click event attributed to this checkout
click_timestamp              Date time in GMT of the click attributed to this checkout
===========================  ===========

Returns
^^^^^^^
The Returns File contains information about returns processed on that day from orders (checkouts) attributed

> to the publisher at specific merchant partners in the past. This report operates at a day level and captures the

Reviewer: "at a day level" reads to me as: the level of each row is a day, so propose to remove since the "captures ..." seems good

```suggestion
to the publisher at specific merchant partners in the past. This report captures the
```

returns processed on a particular day, regardless of when the original checkout occurred. The returns can be partial
or full, and can be broken over multiple days.

The report contains unique rows for each product returned in the order as they are processed, even if multiple products from the same checkout are returned and processed separately.
Each row will include information about the number of items returned(quantity), the amount of earnings borne from the items returned, and the revenue associated with the returned items.
Since the data in these reports pertains to returns, the earnings, revenue and quantity values are negative.

To gain a comprehensive view of returns, we recommend aggregating the values for each unique combination of
checkout_id, merchant and product_id (denoted by the order_product_id column) across all return files.

> Since this report is designed to be used to complement the data in the orders reports, the column 'order_product_id'

Reviewer 2: Should it be a combination of checkout_id and order_product_id to join

Author: The event_product_id is a uniquely generated id for each entry in the events_products table so it should be unique to the combination of merchant+checkout_id+product_id.
